tmt/steps/provision/vagrant.py

```diff
@@ -111,27 +277,34 @@ def destroy(self):
             except GeneralError:
                 sleep(5)
 
+    def ansible(self, playbook):
+        """ Prepare guest using ansible playbook """
+        self.prepare("ansible", playbook)
+
     def prepare(self, how, what):
-        """ add single 'preparator' and run it """
+        """ Add single 'preparation' and run it.
+        """
 
-        name = 'prepare'
+        self.preparations += 1
+        name = 'prepare_' + self.preparations
         cmd = 'provision'
 
         self.vf_backup("Prepare")
 
         # decide what to do
         if how == 'ansible':
-            name = how
+            what = self._ansible_playbook_path(what)
 
```

Comment:
Depends on the context.

```python
def prepare(self, how, what):
```

It handles abstractly multiple "prepare" methods, therefore in the same way. It can be cookbook / shell script / playbook /... anything vagrant and its plugins understand. It can be also path to file, url, or a scriplet...

What this method aims to do is simply pass all what's needed to "Vagrantfile"; in vagrant-speak it's called [Provisioning](https://www.vagrantup.com/docs/provisioning) (see list on the left).

I'll add description to the method.